lib/pbench/server/api/resources/datasets_metadata.py

```diff
@@ -131,7 +131,7 @@ def _put(self, params: ApiParams, req: Request, context: ApiContext) -> Response
         dataset = params.uri["dataset"]
         metadata = params.body["metadata"]
 
-        context["auditing"]["attributes"] = {"updated": metadata}
+        context["auditing"].add_attribute("updated", metadata)
```

Comment:
I wonder whether it's worth having the second entrypoint for `add_attribute()` when you could just do this:
```python
        context["auditing"].add_attributes({"updated": metadata})
```